src/batsim.c

```diff
@@ -0,0 +1,67 @@
+/// \file   batsim.c
+/// \author Bernardo Marques
+/// \date   2023-05
+///
+/// \brief  Battery Simulator - simulates a battery charging
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "include/batsim.h"
+
+#define BAT_C              (910)                 // battery capacitiy in Ah
+#define BAT_VOLTAGE        (12)                  // battery nominal voltage in V
+#define BAT_CAPACITY       (BAT_C * BAT_VOLTAGE) // battery capacity in Wh
+
+#define BAT_MIN_VOLTAGE    (11 * 100)    // battery voltage when fully discharged in V * 100
```

Comment:
We use V*100 so we don't lose precision with the numbers?

Reply:
yes.